01_slides/05_strings.ipynb

```diff
@@ -609,6 +613,256 @@
+              "0"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# count the 'e's\n",
+        "'This string is unusual'.count('e')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "9e016ebe",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "True"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# check if a string ends with a substring\n",
+        "'file_name.csv'.endswith('.csv')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "a9beef6b",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "'long_file_name_with_spaces.csv'"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "# replace text\n",
+        "'long file name with spaces.csv'.replace(' ', '_')"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "9fefba25",
+      "metadata": {},
+      "source": [
+        "For a full list of string methods, see the [Python documentation](https://docs.python.org/3/library/stdtypes.html#string-methods)."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "5e237cbd",
+      "metadata": {},
+      "source": [
+        "### Formatting Strings\n",
+        "\n",
+        "The `format()` string method makes it easier to combine text with other data. We can create text templates with curly braces (`{}`), then pass values in."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "5ae0487a",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "\"Ada Lovelace's initials are A. L.\""
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "'Ada Lovelace\\'s initials are {}. {}.'.format('Ada'[0], 'Lovelace'[0])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "81ea999f",
+      "metadata": {},
+      "source": [
+        "If we preface the string with `f`, we can pass values directly into the curly braces."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "05e7baca",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "\"Ada Lovelace's initials are A. L.\""
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "f'Ada Lovelace\\'s initials are {\"Ada\"[0]}. {\"Lovelace\"[0]}.'"
```

Comment:
Should we explain what `f` is? Should we link a "bonus reading"?

Reply:
This has been modified in the slides.

- **f-string**: `f'...'` is an f-string, it allows for more concise and readable string interpolation by directly embedding expressions within the string.
- **`.format()` method**: This method allows for string formatting by specifying placeholders `{}` within the string and providing values to replace those placeholders in the `.format()` method.

Both expressions essentially achieve the same outcome: they insert the first characters of the `first_name` and `last_name` variables into the string to state Ada Lovelace's initials. The choice between them often comes down to personal preference and the Python version being used. Many developers prefer f-strings for their simplicity and readability.